web/js/modules/product-picker/reducers.js

```diff
@@ -1,42 +1,186 @@
-  filterByAvailable: true,
-  listScrollTop: 0,
+  categoryType: 'hazards and disasters',
+  filters: [],
+  showMobileFacets: true,
+  searchTerm: '',
   selectedLayer: undefined,
   selectedMeasurement: undefined,
   selectedMeasurementSourceIndex: 0,
+  searchConfig: undefined,
+  collapsedFacets: {},
 };
 
 export function getInitialState(config) {
-  return { ...productPickerState, categoryType: Object.keys(config.categories)[1] };
+  return productPickerState;
 }
 
 export function productPickerReducer(state = productPickerState, action) {
   switch (action.type) {
-    case UPDATE_PRODUCT_PICKER:
-      return { ...state, ...action.value };
-    case UPDATE_LIST_SCROLL_TOP:
-      return { ...state, listScrollTop: action.value };
+    case INIT_SEARCH_STATE: {
+      const { searchConfig } = action;
+      return {
+        ...state,
+        searchConfig,
+      };
+    }
+
+    case SAVE_SEARCH_STATE: {
+      const { filters, searchTerm } = action;
+      return {
+        ...state,
+        filters,
+        searchTerm,
+        searchConfig: null,
+      };
+    }
+
+    case COLLAPSE_FACET: {
+      const { field } = action;
+      const { collapsedFacets } = state;
+      return {
+        ...state,
+        collapsedFacets: {
+          ...state.collapsedFacets,
+          [field]: !collapsedFacets[field],
+        },
+      };
+    }
+
+    case SELECT_CATEGORY: {
+      googleTagManager.pushEvent({
+        event: 'layers_meta_category',
+        layers: {
+          meta_category: action.value,
+        },
+      });
+      return {
+        ...state,
+        mode: 'category',
+        categoryType: action.value,
+        selectedMeasurement: null,
+      };
+    }
+
+    case SELECT_MEASUREMENT: {
+      return {
+        ...state,
+        selectedMeasurement: action.value !== state.selectedMeasurement ? action.value : null,
+        selectedMeasurementSourceIndex: 0,
+      };
+    }
+
+    case SELECT_SOURCE: {
+      return {
+        ...state,
+        selectedMeasurementSourceIndex: action.value,
+      };
+    }
+
+    case SELECT_LAYER: {
+      return {
+        ...state,
+        selectedLayer: action.value,
+      };
+    }
+
+    case SHOW_MEASUREMENTS: {
+      const { category, selectedMeasurement } = action.value;
+      googleTagManager.pushEvent({
+        event: 'layers_category',
+        layers: {
+          category: category.title,
+        },
+      });
+      return {
+        ...state,
+        mode: 'measurements',
+        category,
+        selectedMeasurement,
+      };
+    }
+
+    case TOGGLE_SEARCH_MODE: {
+      return {
+        ...state,
+        mode: 'search',
+        selectedLayer: null,
+        showMobileFacets: true,
+        category: null,
+        categoryType: 'hazards and disasters',
+        selectedMeasurementSourceIndex: 0,
+      };
+    }
+
+    case TOGGLE_FEATURED_TAB: {
+      const { config } = action;
+      const category = config.categories.featured.All;
```

Comment:
Maybe a getter here?

Reply:
Likes this?  Or something else:
```

const getSelectedMeasurementId = (config) => {
  const category = config.categories.featured.All;
  const selectedMeasurement = category.measurements[0];
  return config.measurements[selectedMeasurement].id;
}
  
switch ...
  ...
  case TOGGLE_FEATURED_TAB: {
    const { config } = action;
    
    return {
      ...state,
      categoryType: 'featured',
      category,
      mode: 'measurements',
      selectedMeasurement: getSelectedMeasurementId(config),
    };
  }
```

Reply:
I was thinking something like:

`lodashGet(config, 'categories.featured.All');`

So that if `config.categories.featured` isn't there for some reason (someone using a different config), it won't break. But i guess if you are pressing the featured tab it should be there... So probably not necessary.

Reply:
Ah, gotcha